extensions/wikia/Search/UnifiedSearchIndexingController.class.php

```diff
@@ -49,36 +49,52 @@ private function findResults( array $namespaces, $offset, $limit ) {
 				->LIMIT( $limit + 1 )// fetching one more than limit to get next offset
 				->ORDER_BY( [ 'page.page_id', 'asc' ] )
 				->runLoop( $db, function ( &$result, $row ) {
-					$result[] = $row->page_id;
+					$result[] = [ 'id' => $row->page_id ];
 				} );
 
 		$splitByLimit = array_chunk( $results, $limit );
 
+		$items = isset($splitByLimit[0]) ? $splitByLimit[0] : [];
+
 		return [
-			'nextOffset' => isset( $splitByLimit[1][0] ) ? $splitByLimit[1][0] : null,
-			'pageIds' => isset( $splitByLimit[0] ) ? $splitByLimit[0] : [],
+			'nextOffset' => isset( $splitByLimit[1][0]['id'] ) ? $splitByLimit[1][0]['id'] : null,
+			'pageIds' => array_map(function (array $item) {
+				return $item['id'];
+			}, $items),
+			'items' => $items,
 		];
 	}
 
-	private function findWikiResults( $offset, $limit ) {
+	private function findWikis( $offset, $limit ) {
 		$db = wfGetDB( DB_SLAVE, null, 'wikicities' );
 
 		$results =
-			( new WikiaSQL() )->SELECT( 'wikicities.city_list.city_id' )
+			( new WikiaSQL() )->SELECT( 'wikicities.city_list.city_id',
+				'wikicities.city_list.city_lang' )
 				->FROM( 'wikicities.city_list' )
 				->WHERE( 'wikicities.city_list.city_id' )
 				->GREATER_THAN_OR_EQUAL( $offset )
+				->AND_( 'wikicities.city_list.city_public' )
+				->EQUAL_TO( 1 )
 				->LIMIT( $limit + 1 )// fetching one more than limit to get next offset
 				->ORDER_BY( [ 'wikicities.city_list.city_id', 'asc' ] )
 				->runLoop( $db, function ( &$result, $row ) {
-					$result[] = $row->city_id;
+					$result[] = [
+						'id' => $row->city_id,
+						'language' => $row->city_lang,
+					];
 				} );
 
 		$splitByLimit = array_chunk( $results, $limit );
 
+		$items = isset($splitByLimit[0]) ? $splitByLimit[0] : [];
+
 		return [
-			'nextOffset' => isset( $splitByLimit[1][0] ) ? $splitByLimit[1][0] : null,
-			'wikiIds' => isset( $splitByLimit[0] ) ? $splitByLimit[0] : [],
+			'nextOffset' => isset( $splitByLimit[1][0]['id'] ) ? $splitByLimit[1][0]['id'] : null,
+			'wikiIds' => array_map(function (array $item) {
+				return $item['id'];
+			}, $items),
```

Comment:
Should be spacey per coding conventions:
```suggestion
			}, $items ),
```